### PR TITLE
Adding coverage report

### PR DESCRIPTION
**Pull request by arsen-es**
*Issue #, if available:* #17 

*Description of changes:* Migrating Show clause tests, plus, adding jacoco coverage report.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by arsen-es** [comment]
The coverage change is mostly copied from Alerting plugin, you can check their code for reference if interested.

File: build-tools/sqlplugin-coverage.gradle [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ *  To workaround these we start the cluster with jmx enabled and then use Jacoco's JMX MBean to get the execution data before the
+ *  cluster is stopped and dump it to a file. Luckily our current security policy seems to allow this. This will also probably
+ *  break if there are multiple nodes in the integTestCluster. But for now... it sorta works.
+ */
+apply plugin: 'jacoco'
+
+// Get gradle to generate the required jvm agent arg for us using a dummy tasks of type Test. Unfortunately Elastic's
+// testing tasks don't derive from Test so the jacoco plugin can't do this automatically.
+task dummyTest(type: Test) {
```
Comment (penghuo) [inline build-tools/sqlplugin-coverage.gradle]: why this task is needed?

File: build-tools/sqlplugin-coverage.gradle [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    enabled = false
+    workingDir = file("/") // Force absolute path to jacoco agent jar
+    jacoco {
+        destinationFile = file("${buildDir}/jacoco/test.exec")
+        destinationFile.parentFile.mkdirs()
+        jmx = true
+    }
+}
+
+task dummyIntegTest(type: Test) {
```
Comment (penghuo) [inline build-tools/sqlplugin-coverage.gradle]: ditto

File: build-tools/sqlplugin-coverage.gradle (hunk 1 above)
Comment (arsen-es) [reply build-tools/sqlplugin-coverage.gradle]: This one is only used to store the destinationFile setting, we can get rid of this completely and inline that value at the place of usage.

File: build-tools/sqlplugin-coverage.gradle (hunk 2 above)
Comment (arsen-es) [reply build-tools/sqlplugin-coverage.gradle]: This one is used for the jacaco nested object which is converted to jvmArgs in the `integTestCluster`. As for why exactly it is passed in, I think to account for *IT classes' contribution in coverage. I tried building the argument manually, but did not find a good way.

**Pull request merged**